### PR TITLE
[#IR-216] Return the issue tracker url for the tasks in the task list and get API

### PR DESCRIPTION
**Pull request by himanshu-garg-165**

### CODE REVIEW

File: apps/retrospective/serializers/sprint_task.go [hunk 1]
```diff
@@ -6,6 +6,7 @@ import "time"
 type SprintTask struct {
 	ID                   uint
 	Key                  string
+	Url                  string
```
Comment (anmolghosh) [inline apps/retrospective/serializers/sprint_task.go]: What go standards says, `URL` or `Url`? Can you explore a bit

File: apps/retrospective/serializers/retrospective.go [hunk 2]
```diff
@@ -36,3 +37,17 @@ type RetrospectiveCreateSerializer struct {
...
+// GetTaskTrackerConnection ...
+func (retro Retrospective) GetTaskTrackerConnection() (tasktracker.Connection, error) {
+	taskProviderConfig, err := tasktracker.DecryptTaskProviders(retro.TaskProviderConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	connection := tasktracker.GetConnection(taskProviderConfig)
+	if connection == nil {
+		return nil, err
```
Comment (anmolghosh) [inline apps/retrospective/serializers/retrospective.go]: isn't `err` nil here?

File: apps/retrospective/serializers/sprint_task.go [hunk 3]
```diff
@@ -29,7 +30,7 @@ type SprintTask struct {
 
 // SprintTasksSerializer ...
 type SprintTasksSerializer struct {
-	Tasks []SprintTask
+	Tasks []*SprintTask
```
Comment (anmolghosh) [inline apps/retrospective/serializers/sprint_task.go:33]: Hey, help me to understand how this changes the behavior?

File: apps/retrospective/services/sprint_task.go [hunk 4]
```diff
@@ -69,6 +89,22 @@ func (service SprintTaskService) Get(
 		return nil, http.StatusInternalServerError, errors.New("failed to get issue")
 	}
 
+	var retro retroSerializers.Retrospective
```
Comment (anmolghosh) [inline apps/retrospective/services/sprint_task.go]: there is code duplicacy. Please see how can we optimize it

**Review by anmolghosh — CHANGES_REQUESTED** [review]

File: apps/retrospective/models/retrospective.go [hunk 5]
```diff
@@ -90,3 +92,26 @@ func RetroJoinTasks(db *gorm.DB) *gorm.DB {
...
+
+// GetTaskTrackerConnectionFromRetro ...
+func GetTaskTrackerConnectionFromRetro(db *gorm.DB, retroID string) (tasktracker.Connection, error) {
+	var retro Retrospective
+	if err := db.Model(Retrospective{}).
+		Where("retrospectives.deleted_at IS NULL").
+		Where("id = ?", retroID).
+		Find(&retro).Error; err != nil {
+		utils.LogToSentry(err)
+		return nil, errors.New("couldn't find any retrospective with this ID")
```
Comment (anmolghosh) [inline apps/retrospective/models/retrospective.go]: `couldn't find any retrospective with this ID` what ID?
change to `retrospective with ID <value> not found`

File: apps/retrospective/models/retrospective.go [hunk 6]
```diff
@@ -90,3 +92,26 @@ func RetroJoinTasks(db *gorm.DB) *gorm.DB {
...
+	}
+
+	taskProviderConfig, err := tasktracker.DecryptTaskProviders(retro.TaskProviderConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	connection := tasktracker.GetConnection(taskProviderConfig)
+	if connection == nil {
+		return nil, errors.New("no valid connection found")
```
Comment (anmolghosh) [inline apps/retrospective/models/retrospective.go]: log errors

**Review by anmolghosh — CHANGES_REQUESTED** [review]

File: apps/retrospective/serializers/sprint_task.go (hunk 3 above)
Comment (himanshu-garg-165) [reply apps/retrospective/serializers/sprint_task.go:33]: This is done to allow edits in the Task objects in the list if we iterate over them using for loop.

**Review by anmolghosh — APPROVED** [review]

**Pull request merged**